### PR TITLE
t: Reduce back the test module specific timeout limits

### PR DESCRIPTION
**Pull request by okurz**
After introducing global test timeout scaling factors the test module
specific timeout values should be lowered accordingly.

### CODE REVIEW

File: docs/Contributing.asciidoc [hunk 1]
```diff
@@ -638,6 +638,29 @@ standard output. This makes use of `Test::Output` simple but it should be
...
 quite irritating.
 
+=== Test runtime limits
+
+The test modules use `OpenQA::Test::TimeLimit` to introduce a test module
+specific timeout. Consider lowering the timeout value whenever a test module is
+optimized in runtime. If the timeout is hit the test module normally aborts
+with a corresponding message. Please be aware of the exception when the timeout
+triggers after the actual test part of a test module has finished but not all
+involved processes have finished. In this case the output can look like
```
Comment (Martchus) [inline docs/Contributing.asciidoc:648]: Maybe it is worth mentioning the `END` block here. Even without having any sub processes we observed the behavior when the timeout is reached while the `END` block is executed.

File: docs/Contributing.asciidoc [hunk 2]
```diff
@@ -638,6 +638,29 @@ standard output. This makes use of `Test::Output` simple but it should be
...
+Test Summary Report
+-------------------
+t/my_test.t (Wstat: 14 Tests: 1 Failed: 0)
+  Non-zero wait status: 14
+Files=1, Tests=1,  2 wallclock secs ( 0.03 usr  0.00 sys +  0.09 cusr  0.00 csys =  0.12 CPU)
+Result: FAIL
+```
+
+where "Wstat: 14" and "Non-zero wait status: 14" mean that the test process
+received the "ALARM" signal (signal number 14).
```
Comment (Martchus) [inline docs/Contributing.asciidoc:662]: The signal is actually called `ALRM` (or `SIGALRM`).

File: t/lib/OpenQA/Test/TimeLimit.pm [hunk 3]
```diff
@@ -19,7 +19,10 @@ use Test::Most;
 sub import {
     my ($package, $limit) = @_;
     die "$package: Need argument on import, e.g. use: use OpenQA::Test::TimeLimit '42';" unless $limit;
-    $SIG{ALRM} = sub { BAIL_OUT "test exceeds runtime limit of '$limit' seconds\n" };
+    $limit *= $ENV{OPENQA_TEST_TIMEOUT_SCALE_COVER} // 3 if Devel::Cover->can('report');
+    $limit *= $ENV{OPENQA_TEST_TIMEOUT_SCALE_CI}    // 2 if $ENV{CI};
+    print "\$limit: $limit\n";
+    $SIG{ALRM} = sub { print "test exceeds runtime limit of '$limit' seconds\n" };
```
Comment (Martchus) [inline t/lib/OpenQA/Test/TimeLimit.pm:25]: Was the change from `print` to `BAIL_OUT` intended? I'd actually keep the `BAIL_OUT`.

Comment (okurz) [reply t/lib/OpenQA/Test/TimeLimit.pm:25]: yes, definitely. Fixed.

**Comment by okurz** [comment]
Please see https://github.com/os-autoinst/openQA/pull/3419 first . I included your suggestions there. This is a followup PR

File: t/api/04-jobs.t [hunk 4]
```diff
@@ -23,7 +23,7 @@ use File::Temp;
 use Test::Mojo;
 use Test::Output;
 use Test::Warnings ':report_warnings';
-use OpenQA::Test::TimeLimit '180';
+use OpenQA::Test::TimeLimit '20';
```
Comment (kalikiana) [inline t/api/04-jobs.t:26]: This seems to be too little (as seen on this PR's CI results)

Comment (okurz) [reply t/api/04-jobs.t:26]: this multiplies to 120s. I don't see a problem with current CI results. However, you might have found an earlier failed job when I still had 10s timeout

**Review by Martchus — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Pull request merged**